euphonic/spectra.py

```diff
@@ -695,6 +794,114 @@ def broaden(self, x_width: Quantity, shape: str = 'gauss') -> SC:
             deepcopy((self.x_tick_labels)),
             deepcopy(self.metadata))
 
+    def group_by(self, line_data_keys: Union[str, Sequence[str]]) -> SC:
```

Comment:
This would change the calling syntax of `group_by` from `group_by(['key1', 'key2])` to `group_by('key1', 'key2')` but I think that's actually better, and more consistent with the `select(key1=val1, key2=val2)` syntax.... I'll change it